ext/solaris/smf/svc-puppetd

```diff
@@ -12,14 +12,20 @@ exec_prefix=/opt/csw
 sysconfdir=/opt/csw/etc
 sbindir=/opt/csw/sbin
 
-pidfile=/var/run/puppetlabs/agent.pid
+if [ -z $SMF_SYSVOL_FS ]; then 
+  piddir=/var/run
```

Comment:
So in the case this ends up being /var/run/agent.pid, should we also make the pid filename more specific, e.g. puppet-agent.pid? agent.pid is a bit too generic and seems like it could conflict with other software we're not aware of.